symengine/dense_matrix.cpp

```diff
@@ -1613,6 +1631,25 @@ void cross(const DenseMatrix &A, const DenseMatrix &B, DenseMatrix &C)
 
 RCP<const Set> eigen_values(const DenseMatrix &A)
 {
+    unsigned n = A.nrows();
+    bool is_upper = true, is_lower = true;
+    for (unsigned i = 0; i < n; ++i) {
+        for (unsigned j = 0; j < n; ++j) {
+            if (i > j and not eq(*(A.get(i, j)), *zero))
+                is_lower = false;
+            else if (i < j and not eq(*(A.get(i, j)), *zero))
+                is_upper = false;
+        }
+    }
+
+    if (is_lower or is_upper) {
+        RCP<const Set> eigenvals = emptyset();
+        for (unsigned i = 0; i < n; ++i) {
+            eigenvals = set_union({eigenvals, finiteset({A.get(i, i)})});
```

Comment:
You can create one `finiteset` using all of the eigenvalues instead of creating a `finiteset` incrementally.

Reply:
How can that be done? I tried working with finite set and this seemed to be the only way I could make it work

Reply:
`finiteset` takes in a C++ set `set_basic`. You can create a `set_basic` first and then pass that to `finiteset`.

Reply:
Thanks, I will push the changes in a few minutes